packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,685 @@
+      throw new Error('Execution role cannot be undefined when the virtual cluster ID is not a concrete value. Provide an execution role with the correct trust policy');
+    }
+
+    this.logGroup = this.assignLogGroup();
+    this.logBucket = this.assignLogBucket();
+    this.role = this.props.executionRole ?? this.createJobExecutionRole();
+    this.grantPrincipal = this.role;
+
+    this.grantMonitoringPolicies();
+
+    this.taskPolicies = this.createPolicyStatements();
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('emr-containers', 'startJobRun', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        VirtualClusterId: this.props.virtualCluster.id,
+        Name: this.props.jobName,
+        ExecutionRoleArn: this.role.roleArn,
+        ReleaseLabel: this.props.releaseLabel.label,
+        JobDriver: {
+          SparkSubmitJobDriver: {
+            EntryPoint: this.props.jobDriver.sparkSubmitJobDriver?.entryPoint.value,
+            EntryPointArguments: this.props.jobDriver.sparkSubmitJobDriver?.entryPointArguments?.value,
+            SparkSubmitParameters: this.props.jobDriver.sparkSubmitJobDriver?.sparkSubmitParameters,
+          },
+        },
+        ConfigurationOverrides: {
+          ApplicationConfiguration: cdk.listMapper(this.applicationConfigPropertyToJson)(this.props.applicationConfig),
+          MonitoringConfiguration: {
+            CloudWatchMonitoringConfiguration: this.logGroup ? {
+              LogGroupName: this.logGroup.logGroupName,
+              LogStreamNamePrefix: this.props.monitoring!.logStreamNamePrefix,
+            } : undefined,
+            PersistentAppUI: (this.props.monitoring?.persistentAppUI === false)
+              ? 'DISABLED'
+              : 'ENABLED',
+            S3MonitoringConfiguration: this.logBucket ? {
+              LogUri: this.logBucket.s3UrlForObject(),
+            } : undefined,
+          },
+        },
+        Tags: this.props.tags,
+      }),
+    };
+  }
+
+  /**
+   * Render the EMR Containers ConfigurationProperty as JSON
+   */
+  private applicationConfigPropertyToJson = (property: ApplicationConfiguration) => {
+    return {
+      Classification: cdk.stringToCloudFormation(property.classification.classificationStatement),
+      Properties: property.properties ? cdk.objectToCloudFormation(property.properties) : undefined,
+      Configurations: property.nestedConfig ? cdk.listMapper(this.applicationConfigPropertyToJson)(property.nestedConfig) : undefined,
+    };
+  }
+
+  private validateAppConfigPropertiesLength(appConfig?: ApplicationConfiguration) {
+    if (appConfig?.properties === undefined) {
+      return;
+    } else if (Object.keys(appConfig.properties).length > 100) {
+      throw new Error(`Application configuration properties must have 100 or fewer entries. Received ${Object.keys(appConfig.properties).length}`);
+    }
+  }
```

Comment:
` const len = appConfig.properties ?? Object.keys(appConfig.properties).length : 0;` won't work. Did you perhaps mean using the ternary conditional operator instead of the nullish coalescing operator?

Reply:
:) yes I meant to do the ternary operator, sorry!